undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ServletSecurityContextImpl.java

```diff
@@ -314,15 +315,15 @@ static class IdentityContainer implements Serializable {
 
         private static final long serialVersionUID = 812605442632466511L;
 
-        private volatile SecurityIdentity securityIdentity;
+        private volatile CachedIdentity securityIdentity;
         private volatile String authType;
```

Comment:
Can these fields be final?

Reply:
I updated the fields to  be `final` instead of volatile